astropy/io/fits/connect.py

```diff
@@ -140,14 +146,27 @@ def read_table_fits(input, hdu=None, astropy_native=False):
     # Check if table is masked
     masked = False
     for col in table.columns:
+        if include_columns is not None and col.name not in include_columns:
+            continue
         if col.null is not None:
             masked = True
             break
 
     # Convert to an astropy.table.Table object
     # Note: in future, it may make more sense to do this column-by-column,
     # rather than via the structured array.
-    t = Table(table.data, masked=masked)
+    if include_columns is None:
+        t = Table(table.data, masked=masked)
+    else:
+        coldct = {}
```

Comment:
Should be an `OrderedDict` so the output table names are in the same order as `include_columns`.  (And I agree with @saimn that `include_names` would be consistent with `io.ascii`.

(You could also use a list and then init the Table with `names=include_names`).